packages/@expo/cli/src/utils/telemetry/clients/FetchDetachedClient.ts

```diff
@@ -0,0 +1,59 @@
+import { spawn } from 'node:child_process';
+import fs from 'node:fs';
+import path from 'node:path';
+
+import { createTempFilePath } from '../../createTempPath';
+import type { TelemetryClient, TelemetryClientStrategy, TelemetryRecordInternal } from '../types';
+
+const debug = require('debug')('expo:telemetry:client:detached') as typeof console.log;
+
+export class FetchDetachedClient implements TelemetryClient {
+  /** This client should be used for short-lived commands */
+  readonly strategy: TelemetryClientStrategy = 'detached';
+  /** All recorded telemetry events */
+  private records: TelemetryRecordInternal[] = [];
+
+  abort() {
+    return this.records;
+  }
+
+  record(record: TelemetryRecordInternal[]) {
+    this.records.push(
+      ...record.map((record) => ({
+        ...record,
+        originalTimestamp: record.sentAt,
+      }))
+    );
+  }
+
+  async flush() {
+    try {
+      if (!this.records.length) {
+        return debug('No records to flush, skipping...');
+      }
+
+      const file = createTempFilePath('expo-telemetry.json');
+      const data = JSON.stringify({ records: this.records });
+
+      this.records = [];
+
+      await fs.promises.mkdir(path.dirname(file), { recursive: true });
+      await fs.promises.writeFile(file, data);
+
+      const child = spawn(process.execPath, [require.resolve('./flushFetchDetached'), file], {
```

Comment:
nit since I don't have a better comment on this :sweat_smile: Once we bundle this seems a little problematic to solve for, if we then don't create a separate entrypoint. That's probably fine, though probably worth leaving a note here

Also, would it be clearer to use `fork` here?

Reply:
We've talked about this in Slack, keeping it as is.